src/Glob.php

```diff
@@ -112,6 +112,84 @@ public static function glob($glob, $flags = 0)
     }
 
     /**
+     * Combines two globs/paths into another, stricter glob/path.
+     *
+     * @param string $exp1 Path or glob expression.
+     * @param string $exp2 Path or glob expression.
+     *
+     * @return null|string
+     */
+    public static function combine($exp1, $exp2)
```

Comment:
I think we should rename this to `intersect()`, that's easier to understand I guess. Also, please change `$exp1` and `$exp2` to `$glob1` and `$glob2`.